components/product-carousel/product-carousel.tsx

```diff
@@ -0,0 +1,119 @@
+import { removeEdgesAndNodes } from '@bigcommerce/catalyst-client';
+import { useQuery } from '@tanstack/react-query';
+import { forwardRef, useDeferredValue, useId } from 'react';
+
+import {
+  Carousel,
+  CarouselContent,
+  CarouselNextIndicator,
+  CarouselPreviousIndicator,
+  CarouselSlide,
+} from '@bigcommerce/components/carousel';
+
+import { getQuickSearchResults } from '../../client/queries/get-quick-search-results';
+
+import { DummyCard } from './dummy-card';
+import { Pagination } from './pagination';
+import { Product, ProductCard } from './product-card';
+
+interface Props {
+  title: string;
+  categoryId?: number;
+}
+
+type QuickSearchResults = Awaited<ReturnType<typeof getQuickSearchResults>>;
+
+export const fetchQuickSearch = async ({ categoryId }: { categoryId?: number }) => {
+  const response = await fetch(
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
```

Comment:
Maybe you could early return if this is undefined?

Reply:
That completely slipped my mind 🤦‍♂️